sqlglot/dialects/singlestore.py

```diff
@@ -1,5 +1,94 @@
+from sqlglot.dialects.dialect import build_formatted_time
 from sqlglot.dialects.mysql import MySQL
+import typing as t
+from sqlglot import exp, Dialect
+from sqlglot.generator import unsupported_args
+from sqlglot.helper import seq_get
 
 
 class SingleStore(MySQL):
     SUPPORTS_ORDER_BY_ALL = True
+
+    TIME_MAPPING: t.Dict[str, str] = {
+        "AM": "%p",  # Meridian indicator with or without periods
+        "A.M.": "%p",  # Meridian indicator with or without periods
+        "PM": "%p",  # Meridian indicator with or without periods
+        "P.M.": "%p",  # Meridian indicator with or without periods
+        "D": "%u",  # Day of week (1-7)
+        "DD": "%d",  # day of month (1-31)
```

Comment:
`%d` includes zero-padding ([docs](https://docs.python.org/3/library/datetime.html#strftime-strptime-behavior)):

> Day of the month as a zero-padded decimal number.

Should this be `"%-d"`? Singlestore's docs don't show padding:

<img width="376" height="71" alt="Screenshot 2025-07-25 at 5 17 19 PM" src="https://github.com/user-attachments/assets/f296bd6c-729a-450c-83b7-8fac8dc71295" />